firmware/Core/Src/telecommands/i2c_telecommand_defs.c

```diff
@@ -0,0 +1,68 @@
+#include "i2c_telecommand_defs.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "main.h"
+
+uint8_t TCMDEXEC_scan_i2c_bus(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                        char *response_output_buf, uint16_t response_output_buf_len) {
+
+    I2C_HandleTypeDef* hi2c;
+    uint8_t bus_to_scan = atoi(args_str);
```

Comment:
What happens if `args_str` isn't valid? I don't think that's handled here currently

Reply:
What do you mean by not valid? If anything funky is passed to atoi then it will just return 0 and we will handle it as something going wrong in the switch statement.

Reply:
For example, if `args_str = "this is not an integer"`, it's important we consider what happens. Because there's no "i2c 0", that works well enough here.